### DISCUSSION TITLE
Edge weights in Node2Vec

### DISCUSSION THREAD

**Discussion by HaroldGueneau** [pyg-team/pytorch_geometric#3435] [Q&A] [answered]
Hello everyone!

I would like to have a better understanding of the implementation of Node2Vec in PyTorch Geometric.
Specifically, when generating the random walks, where are the edge weights taken into account in the code?

I ask this question especially because I only see the edge indices (edge_index) as input to Node2Vec.

Thanks a lot in advance :)

**Comment by wsad1** [comment, marked as answer]
`Node2Vec` doesn't support `edge_weight` at the moment.

But it was mentioned in an old [roadmap](https://github.com/pyg-team/pytorch_geometric/issues/1465), i'll try to look into it in the coming weeks.

**Comment by HaroldGueneau** [comment]
Ok, thank you for the answer!

**Comment by mustfkeskin** [comment]
is there any progress here?
How we can set edge weights

**Reply by rusty1s** [reply]
We have an open PR about it here: https://github.com/pyg-team/pytorch_geometric/pull/7858

**Comment by thunderock** [comment]
Is there a plan to include this? Not sure if this is still part of pyg.
